packages/test/src/MockResolver.tsx

```diff
@@ -0,0 +1,85 @@
+import {
+  DispatchContext,
+  ReceiveAction,
+  actionTypes,
+  ActionTypes,
+} from '@rest-hooks/core';
+import { useCallback, useContext, useMemo } from 'react';
+import React from 'react';
+
+import { Fixture, actionFromFixture } from './mockState';
+
+type Props = {
+  children: React.ReactNode;
+  fixtures: Fixture[];
+};
+
+/** Can be used to mock responses based on fixtures provided.
+ *
+ * <MockResolver fixtures={postFixtures[state]}><MyComponent /></MockResolver>
+ *
+ * Place below <CacheProvider /> and above any components you want to mock.
+ */
+export default function MockResolver({ children, fixtures }: Props) {
+  const fetchToReceiveAction = useMemo(() => {
+    const actionMap: Record<string, ReceiveAction> = {};
+    for (const fixture of fixtures) {
+      const { key, action } = actionFromFixture(fixture);
+      actionMap[key] = action;
+    }
+    return actionMap;
+  }, [fixtures]);
+
+  const dispatch = useContext(DispatchContext);
+  const dispatchInterceptor = useCallback(
+    (action: ActionTypes) => {
+      if (action.type === actionTypes.FETCH_TYPE) {
+        const { key, resolve, reject } = action.meta;
+        if (key in fetchToReceiveAction) {
```

Comment:
what happens if the key is `'toString'`?
```suggestion
        if (Object.prototype.hasOwnProperty.call(fetchToReceiveAction, key)) {
```

Reply:
key is typed is always be an actual string - so in practice this is less bytes to send over wire (and easier to read).

Reply:
right, but if it's a string that's one of the keys on `Object.prototype`, like `toString`, then this will break.